community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -467,6 +468,17 @@
     public static final Setting<Integer> label_block_size = setting( "unsupported.dbms.block_size.labels", INTEGER,
             "0", min( 0 ) );
 
+    @Description( "Specifies the size of id batches local to each transaction when committing. " +
+            "Committing a transaction which contains changes most often results in new data records being created. " +
+            "For each record a new id needs to be generated from an id generator. " +
+            "It's more efficient to allocate a batch of ids from the contended id generator, which the transaction " +
+            "holds and generates ids from while creating these new records. " +
+            "This setting specifies how big those batches are. " +
+            "Remaining ids are freed back to id generator on clean shutdown." )
+    @Internal
+    public static final Setting<Integer> tx_id_batch_size = setting( "unsupported.dbms.tx_id_batch_size", INTEGER,
+            "20", range( 1, 1_000 ) );
```

Comment:
Maybe `record_id_allocation_batch_size` is more descriptive, since we are not allocating transaction ids.